### PR TITLE
Added a settings panel and some fixes

### PR DESCRIPTION
**Pull request by gnuns**
Please check  #248

### CODE REVIEW

File: js/app.js [hunk 1]
```diff
@@ -34,44 +34,92 @@ var
...
+	}
+	else
+	{
+    	var fs = require('fs');
+    	// The full OS language (with localization, like "en-uk")
+    	var pureLanguage = navigator.language.toLowerCase();
+    	// The global language name (without localization, like "en")
+    	var baseLanguage = navigator.language.toLowerCase().slice(0,2);
+	
+	    if( fs.existsSync('./language/' + pureLanguage + '.json') ) {
```
Comment (jeromelefeuvre) [inline js/app.js:99]: bad tabulation ^_^

File: index.html [hunk 2]
```diff
@@ -138,7 +138,29 @@ <h4 data-translate="movies">Movies</h4>
...
+      	<div class="panel">
+      			
+      		<input id="fullscreenOnStart" type="checkbox">
+			<label for="fullscreenOnStart">Fullscreen on start</label>
+			<br>
+			<input id="closingPrompt" type="checkbox">
+			<label for="closingPrompt">Confirm before closing</label>
+			<br>
+			<input id="alwaysOnFocus" type="checkbox">
+			<label for="alwaysOnFocus">Keep Popcorn Time window always on top</label>
```
Comment (jeromelefeuvre) [inline index.html:153]: Please use locale files

File: index.html [hunk 3]
```diff
@@ -138,7 +138,29 @@ <h4 data-translate="movies">Movies</h4>
...
+				
+			<span>Language: </span>
```
Comment (jeromelefeuvre) [inline index.html:155]: same

File: js/app.js [hunk 4]
```diff
@@ -34,44 +34,92 @@ var
...
+    for(var i in files){
+        if (!files.hasOwnProperty(i)) continue;
+        var name = files[i];
+        if(name.indexOf('.json', name.length - 5) !== -1)
+        {
+            locales.push(name.replace('.json', ''));
+        }
+    }
+   	return locales;
+};
```
Comment (jeromelefeuvre) [inline js/app.js:50]: I'm pretty sure you can add this method in [i18n.js](https://github.com/popcorn-time/popcorn-app/blob/master/node_modules/i18n/i18n.js) module and used it like `i18n.getLocales()`.

File: js/app.js [hunk 5]
```diff
@@ -34,44 +34,92 @@ var
...
-        var key = $el.data('translate');
-
-		if( $el.is('input') ) {
-			$el.attr('placeholder', i18n.__(key));
-		} else {
-			$el.text(i18n.__(key));
-		}
-	});
+	// Check if has a config for app language and if it's valid
+	if(Settings.get('app_language') && getLocales('./language').indexOf(Settings.get('app_language') != -1))
```
Comment (jeromelefeuvre) [inline js/app.js:87]: same here for adding a new function like:

```
i18n.isValid(Settings.get('app_language'))
```

File: js/app.js [hunk 6]
```diff
@@ -93,12 +141,29 @@ var populateCategories = function() {
 
 detectLanguage('en');
 
+// add available langauage options to Settings Panel
```
Comment (jeromelefeuvre) [inline js/app.js:144]: language ;)

File: js/app.js [hunk 7]
```diff
@@ -34,44 +34,92 @@ var
...
+        	Settings.set('app_language', pureLanguage);
+    	}
+    	else if( fs.existsSync('./language/' + baseLanguage + '.json') ) {
+        	i18n.setLocale(baseLanguage);
+        	Settings.set('app_language', baseLanguage);
+    	} else {
+        	i18n.setLocale(preferredLanguage);
+        	Settings.set('app_language', preferredLanguage);
+    	}
+
```
Comment (jeromelefeuvre) [inline js/app.js:110]: you can remove all lines with `Settings.set('app_language', ...)'` and adding l.110 :
`Settings.set('app_language', I18n.getLocale());`

**Comment by gnuns** [comment]
Ok @jeromelefeuvre, thanks for the comments.
I was in doubt if I could change the i18n.js, so...
I will fix all this ;)

**Comment by jeromelefeuvre** [comment]
yes you can, you can fork i18n-node too and add a pull request of your changes too

**Pull request closed without merging**